### PR TITLE
[RFC] Upgrade Rust toolchain to 1.49.0

### PR DESCRIPTION
**Pull request by serban300**
# Reason for This PR

#2434 

## Description of Changes

Upgrade Rust toolchain to 1.49.0

The main highlight of this upgrade is the aarch64-unknown-linux-gnu target Tier 1 support.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

Things worth mentioning:

1. In order to workaround the issue https://github.com/rust-lang/rust/issues/79791 on aarch64 we need to build libfdt manually, appending `-fno-stack-protector` to `CFLAGS`
2. The process startup time increases on aarch64 when using jailer. Maybe because of [this](https://github.com/rust-lang/rust/blob/1.49.0/library/std/src/sys/unix/mod.rs#L118)

If all these issues are acceptable, we can promote this draft to an actual pull request.

### CODE REVIEW

**Comment by acatangiu** [comment]
The regression on the failing block performance tests looks quite big! I propose we drill down a bit there to get more info.

**Comment by serban300** [comment]
> The regression on the failing block performance tests looks quite big! I propose we drill down a bit there to get more info.

@acatangiu Sorry, my bad. I forgot to pin the Firecracker threads before running the block performance tests. Now they pass successfully.

**Comment by acatangiu** [comment]
> @acatangiu Sorry, my bad. I forgot to pin the Firecracker vcpu threads before running the block performance tests. Now they pass successfully.

Phew! 👍 

> The process startup time increases on aarch64 when using jailer. Maybe because of this

What are the numbers? 😄

File: tests/integration_tests/performance/test_process_startup_time.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 import host_tools.logging as log_tools
 
-MAX_STARTUP_TIME_CPU_US = {'x86_64': 5500, 'aarch64': 2800}
+MAX_STARTUP_TIME_CPU_US = {'x86_64': 5500, 'aarch64': 3200}
```
Comment (acatangiu) [inline tests/integration_tests/performance/test_process_startup_time.py:12]: I guess this is fine, 400us is negligible.

**Review by acatangiu — DISMISSED** [review]
LGTM 🎉

**Comment by acatangiu** [comment]
Also, why not latest then? :D https://blog.rust-lang.org/2021/05/10/Rust-1.52.1.html

**Comment by serban300** [comment]
> Also, why not latest then? :D https://blog.rust-lang.org/2021/05/10/Rust-1.52.1.html

Upgrading to the latest Rust toolchain sounds good to me. I'll get back to this item after implementing #2153 .

**Comment by serban300** [comment]
Resolving in favor of #2579

**Pull request closed without merging**